core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -12310,6 +12310,21 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.ORACLE.getDialect()), isLinux(expectedSparkQuery));
   }
 
+  @Test public void testToCharInSpark() {
```

Comment:
Test case should be for HOUR_OF_DAY_12 also. Either update the same test case or add new one.

Reply:
@dm-shantanuc i had added the format.Please have a look.